contracts/LoanCore.sol

```diff
@@ -389,34 +436,39 @@ contract LoanCore is
      *         net payments have been collected by the RepaymentController for withdrawal.
      *
      * @param oldLoanId             The ID of the old loan.
+     * @param oldLender             The lender for the old loan.
      * @param borrower              The borrower for the loan.
      * @param lender                The lender for the old loan.
      * @param terms                 The terms of the new loan.
      * @param _settledAmount        The amount LoanCore needs to withdraw to settle.
      * @param _amountToOldLender    The payment to the old lender (if lenders are changing).
      * @param _amountToLender       The payment to the lender (if same as old lender).
      * @param _amountToBorrower     The payment to the borrower (in the case of leftover principal).
+     * @param _interestAmount       The interest amount to be paid.
      *
      * @return newLoanId            The ID of the new loan.
      */
     function rollover(
         uint256 oldLoanId,
+        address oldLender,
         address borrower,
         address lender,
         LoanLibrary.LoanTerms calldata terms,
         uint256 _settledAmount,
         uint256 _amountToOldLender,
         uint256 _amountToLender,
-        uint256 _amountToBorrower
+        uint256 _amountToBorrower,
+        uint256 _interestAmount
     ) external override whenNotPaused onlyRole(ORIGINATOR_ROLE) nonReentrant returns (uint256 newLoanId) {
         LoanLibrary.LoanData storage data = loans[oldLoanId];
         // Ensure valid loan state for old loan
         if (data.state != LoanLibrary.LoanState.Active) revert LC_InvalidState(data.state);
 
         // State change for old loan
         data.state = LoanLibrary.LoanState.Repaid;
+        data.balance = 0;
```

Comment:
Same thing here, just looks like cleanup to make sure rollovers handle proration cleanly right? Not related to partial repayments

Reply:
My intention with these state updates on 469 and 470 were to reflect the rollover 'repayment' in the old loanData to show that it was paid off by the rollover action. I wasnt thinking about the zero balance checks and as mentioned in last comment think it is best to use the LoanState.